### PR TITLE
instancetypes: Address functional test TODOs and extend libdv

### PR DESCRIPTION
**Pull request by lyarwood**
/sig code-quality

### What this PR does

This PR introduces `WithDefault{Instancetype,Preference` and `WithDataVolumeSourceRef` to `libdv` to address some basic `TODOs` within the instancetype functional tests.

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

File: tests/libdv/dv.go [hunk 1]
```diff
@@ -151,6 +154,36 @@ func WithForceBindAnnotation() dvOption {
...
+	return func(dv *v1beta1.DataVolume) {
+		if dv.Labels == nil {
+			dv.Labels = map[string]string{}
+		}
+		dv.Labels[instancetypeapi.DefaultPreferenceLabel] = name
+		dv.Labels[instancetypeapi.DefaultPreferenceKindLabel] = kind
+	}
+}
+
+func WithDataVolumeSourceRef(name, namespace, kind string) dvOption {
```
Comment (0xFelix) [inline tests/libdv/dv.go:177]:
```suggestion
func WithDataVolumeSourceRef(namespace, name, kind string) dvOption {
```

Stay consistent with other options?

https://github.com/kubevirt/kubevirt/blob/51f66d67e52cdcd2f07d4a6df12d65eab423b117/tests/libdv/dv.go#L137

Comment (EdDev) [reply tests/libdv/dv.go:177]: It is a nit, but I also have hard time to ignore it :)
I would even go one step ahead and argue that the `kind` part should come before them all (including the prev commit).

**Review by 0xFelix — APPROVED** [review]
/lgtm

**Pull request merged**